core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -179,13 +181,16 @@ public void indexWeightChanged() {}
     }
 
     this.opts = opts;
+    if (null != opts.tableConfig && opts.tableConfig.size() > 0) {
+      ConfigurationCopy tableCC = new ConfigurationCopy(DefaultConfiguration.getInstance());
+      opts.tableConfig.forEach(tableCC::set);
+      this.tableConf = tableCC;
+    } else {
+      this.tableConf = DefaultConfiguration.getInstance();
+    }
 
     if (opts.indexCacheSize > 0 || opts.dataCacheSize > 0) {
-      ConfigurationCopy cc = new ConfigurationCopy(DefaultConfiguration.getInstance());
-      if (null != opts.tableConfig) {
-        opts.tableConfig.forEach(cc::set);
-      }
-
+      ConfigurationCopy cc = new ConfigurationCopy(this.tableConf);
```

Comment:
Why can't we use `this.tableConf` directly here, since a copy has been made above?

Reply:
I made this.tableConf an AccumuloConfiguration object, which won't allow to set properties for the block cache config.  I can change it back to a ConfigurationCopy object...  I think it would be OK to set the cache properties in the saved object.

Reply:
Could do 

```java
  ConfigurationCopy cc = this.tableConf instanceOf ConfigurationCopy ? (ConfigurationCopy) this.tableConf : new ConfigurationCopy(this.tableConf)
```